Dsd 140 work order route

Dissurender opened the pull request.

hector3211 reviewed (approved):
Just update the changes we talked about but other than that good job!

reematcs reviewed (approved):
LGTM.

musselmandev reviewed (approved):
Learned about patches, that was new to me!

Dissurender commented:
Due to database scripts and generated files, the scope of this PR has changed to affect more files in the project.

- I have modified the migration sql to use postgres `now()` on creations and updates on the database level
- I have refactored multiple files to remove the updated_at and created_at params/fields

sources:
- https://wiki.postgresql.org/wiki/Don't_Do_This#Don.27t_use_timestamp_.28without_time_zone.29
- https://www.tinybird.co/blog-posts/database-timestamps-timezones
- https://stackoverflow.com/questions/72833844/who-is-responsible-for-modifying-timestamp-fields-in-the-app-backend-or-db

reematcs commented:
@Dissurender btw, this line is wrong (it’s out of scope, but it is just logically wrong in the SQL - I changed it on my branch/locally but by the time I am done with my ticket it will persist).
`CREATE INDEX "lease_apartment_id_index" ON leases ("id”);`

should be

`CREATE INDEX "lease_apartment_id_index" ON leases ("apartment_id");`

reematcs commented:
Sigh, I just deleted something I was right about.

This line:

`CREATE INDEX "lease_apartment_id_index" ON leases ("id");`

Should be

`CREATE INDEX "lease_apartment_id_index" ON leases ("apartment_id");`

It’s out of scope, but it will persist until I merge. Please check it @Dissurender

reematcs commented:
@Dissurender please check this line:

`"lease_apartment_id_index" ON leases (“id”)`

hector3211 commented on backend/internal/db/migrations/000001_init.up.sql:
```diff
@@ -47,57 +48,57 @@ CREATE TYPE "Work_Category" AS ENUM (
+    "unit_number" SMALLINT         NULL,
+    "role"        "Role"           NOT NULL DEFAULT "Role" 'tenant',
+    "status"      "Account_Status" NOT NULL DEFAULT "Account_Status" 'active',
+    "last_login"  TIMESTAMP(0)     NOT NULL,
```
We can drope "last_loging" not really needed at this point and we can call clerk to get this.

hector3211 commented on backend/server.go:
```diff
@@ -243,6 +243,36 @@ func main() {
+		})
+
+		// All route
+		r.Post("/", func(w http.ResponseWriter, r *http.Request) {
```
Lets do "/create" or "/new" for this Post request also making it more readable

hector3211 commented on backend/internal/db/queries/users.sql:
```diff
@@ -6,21 +6,19 @@ INSERT INTO users (
-    last_login,
-    updated_at,
-    created_at
+    last_login
```
Remove this "last_login" as well Clerk handles this
